packages/langium/test/lsp/find-references.test.ts

```diff
@@ -0,0 +1,132 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { createLangiumGrammarServices } from '../../src';
+import { expectFindReferences } from '../../src/test';
+import { expectFunction } from '../fixture';
+
+const text = `
+grammar test hidden(WS)
+
+entry Model: value=RuleA;
+
+interface <|>A {
+    na<|>me: string
+}
+
+interface B extends A {
+    assignmentB: string
+}
+
+type C = A | B;
+
+RuleA returns A: na<|>me=ID;
+
+RuleB returns B: na<|>me=ID assignmentB=ID;
+
+RuleC returns C: na<|>me=ID assi<|>gnmentB=ID;
+
+ActionRule: {A} 'A' na<|>me=ID | {B} name=ID assig<|>nmentB=ID | {C} na<|>me=ID assignmentB=ID;
+
+terminal ID: /\\w+/;
+terminal WS: /\\s+/;
+
+`;
+
+const grammarServices = createLangiumGrammarServices().grammar;
+const findReferences = expectFindReferences(grammarServices, expectFunction);
+
+describe('findReferences', () => {
+    test('Must find all references to interface A from interface\'s name including declaration', async () => {
+        await findReferences({
```

Comment:
I know this pattern from goto and hover. I have some doubts to this form, because it hides what is tested. Sure, I can go to the definition of the `text` variable or to this `findReferences` definition. But you have to do much effort to find out what is done.
Suggestions:
* break out every `findReferences` into three parts (3 A's):
  * one for preparing the situation (arrange)
  * one for doing the action that is important for the test case (act)
  * one for asserting the state after this action (assert)
* also split the giant grammar text into small texts in each test case
  * because to understand one case now you will have to scroll up and count down the index of your range, which is really annoying I would say...

Reply:
After our discussion, I assigned a grammar text for each test. This (I hope) makes the test easier to read.

The tests now check that the length of the returned array is correct **and** that the range of the reference is also correct.